nni/algorithms/compression/v2/pytorch/pruning/tools/base.py

```diff
@@ -375,10 +375,15 @@ def _expand_mask(self, name: str, mask: Tensor) -> Dict[str, Tensor]:
         weight_mask = mask.clone()
 
         if self.block_sparse_size is not None:
+            block_sparse_size = list(self.block_sparse_size)
+            # padding block_sparse_size to weight_mask length if dim is None
+            if self.dim is None:
+                for _ in range(len(weight_mask.size()) - len(self.block_sparse_size)):
```

Comment:
Do you mean block size [2, 2] align with weight shape [5, 5]? we didn't support this situation, it will raise error😂
This part is aligning block size [2, 2] with weight shape [4, 4, 4], and will padding block size to [1, 2, 2].
The reminding is in the params docstr.